From d03a041790e1d0ecb900ec1de57580bca9519daa Mon Sep 17 00:00:00 2001
From: Mark Wielaard <mark@klomp.org>
Date: Sat, 2 Jul 2022 01:11:00 +0200
Subject: [PATCH] Redirect stder in gdb-add-index.sh test

gdb-add-index might produce an error message on stderr when trying to
disable debuginfod support. Any message to stderr makes the testcase
fail. This looks like a gdb bug:
https://sourceware.org/bugzilla/show_bug.cgi?id=29316
But it is easy to workaround by redirecting stderr to stdout.
---
 testsuite/dwz.tests/gdb-add-index.sh | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/testsuite/dwz.tests/gdb-add-index.sh b/testsuite/dwz.tests/gdb-add-index.sh
index 5a91b23..3095efb 100644
--- a/testsuite/dwz.tests/gdb-add-index.sh
+++ b/testsuite/dwz.tests/gdb-add-index.sh
@@ -1,6 +1,8 @@
 cp $execs/hello 1
 
-gdb-add-index 1
+# Redirect gdb-add-index stderr to stdout.
+# https://sourceware.org/bugzilla/show_bug.cgi?id=29316
+gdb-add-index 1 2>&1
 
 readelf -S 1 | grep -q '\.gdb_index'
 
-- 
2.31.1

